pages/index.js

```diff
@@ -1,5 +1,47 @@
-import CategoryAnimals from "../components/CardSection/CategoryAnimals";
+import CategoryOverview from "../components/CategoryOverview/CategoryOverview";
+import Header from "../components/Header/Header";
+import Link from "next/link";
+import { words } from "../dummydata/words";
+import { nanoid } from "nanoid";
+import { Fragment } from "react";
+import styled from "styled-components";
+
+const allCategories = words.map((word) => word.category);
+const uniqueCategories = Array.from(new Set(allCategories));
+
+const singleCategories = uniqueCategories.map((category) => {
+  return (category = words.filter((word) => word.category === category));
+});
+
+export const wordsInCategories = singleCategories.map((item) => {
+  return {
+    id: nanoid(),
+    slug: item[0].category,
```

Comment:
I think item is already ONE Object, you dont need the index here, as far as I understand your Data.